src/plots/cartesian/axes.js

```diff
@@ -1604,7 +1604,7 @@ axes.doTicks = function(gd, axid, skipTitle) {
     var axletter = axid.charAt(0),
         counterLetter = axes.counterLetter(axid),
         vals = axes.calcTicks(ax),
-        datafn = function(d) { return d.text + d.x + ax.mirror; },
+        datafn = function(d) { return [d.text, d.x, ax.mirror].join('\u0000'); },
```

Comment:
Ok. In this case, I would prefer changing the `join` argument to `-` if you don't mind.